src/network/websockets/server/index.ts

```diff
@@ -44,7 +45,32 @@ export function runWebsocketServer<AuthDetails>(params: Params<AuthDetails>) {
       server: params.server
     });
 
-  server.on("connection", function(socket) {
+  function heartbeat(this: any) {
+    this.lifeFlag = "good";
+  }
+  setInterval(function ping() {
+    server.clients.forEach(
+      (socket: WebSocket & { lifeFlag?: "good" | "sleepy" }) => {
```

Comment:
Just throwing out some other terms if they sound better:
`stalled`
`alive`, `dead`
`online`, `offline`
`active`, `inactive`

Reply:
The algorithm in this PR marks every socket as sleepy unconditionally and then it’s then the job of the socket to jump up and say "[I’m not dead](https://www.youtube.com/watch?v=uBxMPqxJGqI)".

https://github.com/citrusbyte/oncilla/blob/4f9d7667279ceabf8a470d7dc6da9faa5c31cbed/src/network/websockets/server/index.ts#L60-L61

If a socket was marked dead, offline, or inactive, why wouldn’t we terminate the socket right away? So this temporary state of a socket is sleepy, testing, questionable, unknown. I’m happy to get a PR with any rename of that sort!